azurerm/internal/services/applicationinsights/application_insights_webtests_resource.go

```diff
@@ -237,13 +237,13 @@ func resourceArmApplicationInsightsWebTestsRead(d *schema.ResourceData, meta int
 	d.Set("application_insights_id", appInsightsId)
 	d.Set("name", resp.Name)
 	d.Set("resource_group_name", resGroup)
-	d.Set("kind", resp.Kind)
 
 	if location := resp.Location; location != nil {
 		d.Set("location", azure.NormalizeLocation(*location))
 	}
 
 	if props := resp.WebTestProperties; props != nil {
+		d.Set("kind", string(props.WebTestKind))
```

Comment:
This would maybe be safer checking for the top-level field first?

```suggestion
		if resp.Kind == "" {
			d.Set("kind", string(props.WebTestKind))
		}
```